.circleci/config.yml

```diff
@@ -0,0 +1,82 @@
+version: 2
+
+jobs:
+  python3:
+    docker:
+      - image: circleci/python:3.6.1
+    environment:
+      - MINICONDA_PATH: ~/miniconda
+      - CONDA_ENV_NAME: testenv
+      - PYTHON_VERSION: 3
+      - MATPLOTLIB_VERSION: "*"
+    steps:
+      - checkout
+      - run: ./build_tools/circle/checkout_merge_commit.sh
+      - restore_cache:
+          key: v1-datasets-{{ .Branch }}
+      - run: ./build_tools/circle/build_doc.sh
+      - save_cache:
+          key: v1-datasets-{{ .Branch }}
+          paths:
+            - ~/scikit_learn_data
+      - store_artifacts:
+          path: doc/_build/html/stable
+          destination: doc
+      - store_artifacts:
+          path: ~/log.txt
+          destination: log.txt
+      - persist_to_workspace:
+          root: doc/_build/html
+          paths: .
+
+
+  python2:
+    docker:
+      - image: circleci/python:3.6.1
+    environment:
+      - MINICONDA_PATH: ~/miniconda
+      - CONDA_ENV_NAME: testenv
+      - PYTHON_VERSION: 2
+      - MATPLOTLIB_VERSION: "*"
+    steps:
+      - checkout
+      - run: ./build_tools/circle/checkout_merge_commit.sh
+      - restore_cache:
+          key: v1-datasets-{{ .Branch }}-python2
+      - run: ./build_tools/circle/build_doc.sh
+      - save_cache:
+          key: v1-datasets-{{ .Branch }}-python2
+          paths:
+            - ~/scikit_learn_data
+      - store_artifacts:
+          path: doc/_build/html/stable
+          destination: doc
+      - store_artifacts:
+          path: ~/log.txt
+          destination: log.txt
+
+  deploy:
+    docker:
+      - image: circleci/python:3.6.1
+    steps:
+      - checkout
+      - run: ./build_tools/circle/checkout_merge_commit.sh
```

Comment:
why is this needed if we're just pushing the changes from another build?

Reply:
I think this is just easier to always run for all steps on the result of the merge into master of the PR branch.